### PR TITLE
MINOR: Fix some compiler warnings

### PR DESCRIPTION
**Pull request by ijuma**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/server/FetchSessionTest.scala [hunk 1]
```diff
@@ -201,7 +201,7 @@ class FetchSessionTest {
       context6.updateAndGenerateResponseData(respData2).error())
 
     // Close the incremental fetch session.
-    var prevSessionId = resp5.sessionId()
+    var prevSessionId = resp5.sessionId() //FIXME what should happen here?
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/server/FetchSessionTest.scala]: @cmccabe what was intended here, should it be a `val` or was the intent to update `prevSessionId` in the loop?

Comment (cmccabe) [reply core/src/test/scala/unit/kafka/server/FetchSessionTest.scala]: It can be a `val` here.

Comment (cmccabe) [reply core/src/test/scala/unit/kafka/server/FetchSessionTest.scala]: Just to clarify, the purpose of the loop is to verify that if we destroy and later re-create a session, we eventually get a new ID.  Although the chances of an ID collision are extremely low, I didn't want to write a nondeterministic unit test.

Comment (ijuma) [reply core/src/test/scala/unit/kafka/server/FetchSessionTest.scala]: Thanks for the explanation.

**Review by rajinisivaram — APPROVED** [review]
@ijuma Thanks for the PR, LGTM.

**Comment by rajinisivaram** [comment]
retest this please

**Comment by rajinisivaram** [comment]
Merged to trunk

**Pull request merged**